core/src/test/java/hudson/util/AtomicFileWriterTest.java

```diff
@@ -185,12 +189,8 @@ public void checkPermissions() throws IOException, InterruptedException {
 
         final Set<PosixFilePermission> posixFilePermissions = Files.getPosixFilePermissions(filePath);
 
-        for (PosixFilePermission perm : givenPermissions) {
-            assertTrue("missing: " + perm, posixFilePermissions.contains(perm));
-        }
-
-        for (PosixFilePermission perm : notGivenPermissions) {
-            assertFalse("should not be allowed: " + perm, posixFilePermissions.contains(perm));
+        for (PosixFilePermission perm : DEFAULT_GIVEN_PERMISSIONS) {
+            assertTrue("missing expected permission: " + perm, posixFilePermissions.contains(perm));
```

Comment:
Maybe replace the loop with `assertThat(posixFilePermissions, equalTo(DEFAULT_GIVEN_PERMISSIONS))` to catch any changes that add extra permissions as well? You would also get nicely formatted failures for free:

    Expected: <[OWNER_READ, GROUP_READ, OTHERS_READ, OWNER_WRITE]>
         but: was <[OWNER_READ, OTHERS_WRITE, GROUP_READ, OTHERS_READ, OWNER_WRITE]>

Reply:
Done.

Thanks @dwnusbaum !